src/app/components/wrapper-components/demo-card-wrapper/demo-card-wrapper.component.spec.ts

```diff
@@ -81,8 +94,64 @@ describe('DemoCardComponent', () => {
         // Given any starting state of component
         // When passing
         const result: void = await testUtils.getComponent().onCancelMove('not even necessary');
-        // Then nothing should have happend (for coverage sake)
+        // Then nothing should have happen (for coverage sake)
         expect(result).withContext('should be null').toBe();
     }));
 
+    it('should reload node when inputs are updated by parents', fakeAsync(async() => {
+        // Given a component already initialized with one given set of infos
+        loadNode({
+            name: 'P4',
+            node: new GameNode(P4Rules.get().getInitialState(defaultConfig)),
+            click: MGPOptional.empty(),
+        });
+        testUtils.expectElementNotToExist('.player0-fill');
+
+        // When loading another component, which triggers ngOnChanges
+        const boardWithPiece: Table<PlayerOrNone> = TableUtils.create(7, 6, PlayerOrNone.ZERO);
+        const stateWithPieces: P4State = new P4State(boardWithPiece, 42);
+        loadNode({
+            name: 'P4',
+            node: new GameNode(stateWithPieces),
+            click: MGPOptional.empty(),
+        });
+        await testUtils.getComponent().ngOnChanges({} as SimpleChanges);
```

Comment:
Alors, j'ai expérimenté et checké un peu la doc / les commentaires des gens.
Quand tu changes un @Input depuis l'html (genre si c'est un champ texte que tu changes), `ngOnChanges` est appelé. Si tu fais `composant.monInput = 42`, il est *pas* appelé, d'où la nécessité de le triggerer manuellement.
Ce que je propose, pour mieux refléter ça, c'est de bouger cet appel à `ngOnChanges` dans `loadNode`, ainsi on reflétera le vrai comportement dans TOUS les tests de demo-card.
Ça donne la déf suivante, et tous les tests passent avec : 

```
    async function loadNode(nodeInfo: DemoNodeInfo): Promise<void> {
        testUtils.getComponent().demoNodeInfo = nodeInfo;
        await testUtils.getComponent().ngOnChanges({} as SimpleChanges);
        testUtils.detectChanges();
        tick(1); // Need at least 1ms because of the setTimeout in ngAfterViewInit
    }
```